tasks/tasks.go

```diff
@@ -40,24 +57,28 @@ func Sync(ctx context.Context, config *tsbridge.Config) error {
 	}
 	defer store.Close()
 
-	metrics, err := tsbridge.NewMetricConfig(ctx, config, store)
-	if err != nil {
-		return err
-	}
+	metricConfigOnce.Do(func() {
+		metricConfig, err = tsbridge.NewMetricConfig(ctx, config, store)
+		if err != nil {
+			log.Fatalf("Unable to initialize metric config: %v", err)
```

Comment:
hmm, I'm not sure log.Fatal is the right call here, as it'll call os.Exit right after, and does not allow for clean closure of open handlers.

Might be cleaner to return an error instead, which allows for handlers such as `store.Close()` etc to be executed.

Same for the other log.Fatal's below.

Reply:
That's a good point. Fixed. Can you PTAL?